Libplanet.Net/BlockCandidateTable.cs

```diff
@@ -18,79 +17,155 @@ namespace Libplanet.Net
+            Branches = new ConcurrentBag<CandidateBranch<T>>();
+            _lock = new object();
         }
 
-        public long Count
-        {
-            get => _blocks.Count;
-        }
+        /// <summary>
+        /// Retrieves the <see cref="CandidateBranch{T}"/> with the best Tip, i.e.
+        /// the highest <see cref="Block{T}.TotalDifficulty"/>.
+        /// </summary>
+        public CandidateBranch<T>? BestBranch { get; private set; }
 
-        public bool Any() => !_blocks.IsEmpty;
+        public long Count => Branches.Count;
 
         /// <summary>
-        /// Adds a <see cref="Block{T}"/>s to the table.
+        /// An <see cref="IEnumerable{T}"/> of
+        /// <see cref="CandidateBranch{T}"/>es that can be appended
+        /// to the <em>current</em> <see cref="BlockChain{T}"/>.
         /// </summary>
-        /// <param name="blockHeader">This is the header of the <see cref="BlockChain{T}"/>
-        /// tip at the time of downloading the blocks.</param>
-        /// <param name="blocks">List of downloaded <see cref="Block{T}"/>.</param>
-        public void Add(BlockHeader blockHeader, IEnumerable<Block<T>> blocks)
-        {
-            if (_blocks.ContainsKey(blockHeader))
-            {
-                return;
-            }
-
-            try
-            {
-                var sortedBlocks =
-                    new SortedList<long, Block<T>>(blocks.ToDictionary(i => i.Index));
-                _blocks.TryAdd(blockHeader, sortedBlocks);
-            }
-            catch (ArgumentException e)
-            {
-                Log.Debug(
-                    "Among the previous blocks of BlockHeader " +
-                    "#{Index} {BlockHash}, there is a block with the wrong index. " +
-                    "InnerMessage: {InnerMessage}",
-                    blockHeader.Index,
-                    blockHeader.Hash,
-                    e.Message);
-            }
-        }
+        private ConcurrentBag<CandidateBranch<T>> Branches { get; set; }
 
         /// <summary>
-        /// Get the <see cref="Block{T}"/>s which are in the table.
+        /// Adds an <see cref="CandidateBranch{T}"/> to <see cref="Branches"/>.
         /// </summary>
-        /// <param name="thisRoundTip">Canonical <see cref="BlockChain{T}"/>'s
-        /// tip of this round.</param>
-        /// <returns><see cref="Block{T}"/>s by <paramref name="thisRoundTip"/>.</returns>
-        public SortedList<long, Block<T>>? GetCurrentRoundCandidate(
-            BlockHeader thisRoundTip)
+        /// <param name="branch">The <see cref="CandidateBranch{T}"/>
+        /// to add to <see cref="Branches"/>.
+        /// </param>
+        public void Add(CandidateBranch<T> branch)
         {
-            return _blocks.TryGetValue(thisRoundTip, out var blocks)
-                ? blocks
-                : null;
-        }
+            BestBranch ??= branch;
+            BestBranch = CompareBranch(BestBranch, branch);
```

Comment:
This is might be the wrong behavior. If the new branch is a reorg branch and branchpoint is missing, `CompareBranch` is cannot return correctly. 

Plus, `Add` is do not have other behavior except `Add`. This is might be a cause of the mistake. (Even do not note this behavior on the docstring.)

Reply:
After removing `BestBranch` comparing, it needs to be call update at some point to get a valid `BestBranch` not `null`. However, the problem is, `Update()` requires a fork path `UpdatePath<T>` from Blockchain. 

If node is up or new Blockchain, `Update()` cannot be called because there's no  `UpdatePath<T>` to use. because `UpdatePath<T>` hasn't created yet.